File: leetcode/exercises/1.3.1.easy.valid_anagram_1.js
// -----------------------
// Created     : 24/10/2025
// Last Edited : 24/10/2025
// Topics      : 
// Big O       :
// Problem Id  : 242
// Source      :
// Notes       :
// -----------------------

// note(@morenoblue): This solution isn't finished. I have no idea how to 
//                    reliably compare two objects in javascript without 
//                    importing a library. But its just a matter of comparing 
//                    the two objects in the end to check if they are the same, 
//                    which if they are, then the strings are indeed anagrams.

/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length !== t.length) { return false };

    s_counter = {};
    t_counter = {};

    for (let i = 0; i < s.length; i++)
    {
       s_counter[s[i]] = (s_counter[s[i]] ?? 0) + 1 
       t_counter[t[i]] = (t_counter[t[i]] ?? 0) + 1 
    };

};

File: leetcode/exercises/1.3.1.easy.valid_anagram_1.py
# -----------------------
# Created     : 20/10/2025
# Last Edited : 24/10/2025
# Topics      : 
# Big O       :
# Problem Id  : 242
# Source      :
# Notes       :
# -----------------------

class Solution:
	def isAnagram(self, s: str, t: str) -> bool:
		if len(s) != len(t):
			return False
		map1, map2 = {}, {}
		for i,j in zip(s,t):
			map1[i] = map1.get(i, 0) + 1
			map2[j] = map2.get(j, 0) + 1

		return map1 == map2
